src/main/java/com/marklogic/kafka/connect/source/SerializedConstraintColumnHandler.java

```diff
@@ -0,0 +1,92 @@
+                ArrayNode rootArgsArray = (ArrayNode) queryRoot.get("$optic").get("args");
+
+                rootArgsArray.insert(1, constraintNode);
+                constrainedQuery = mapper.writeValueAsString(queryRoot);
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        logger.debug("Serialized constrainedQuery (unless initial run): " + constrainedQuery);
+        return constrainedQuery;
+    }
+
+    @Override
+    public String getMaxConstraintValue(String currentQuery, long queryStartTimeInMillis, String constraintColumn) {
+        String currentMaxValueQuery = buildMaxValueSerializedQuery(currentQuery, constraintColumn);
+        logger.debug("currentMaxValueQuery: " + currentMaxValueQuery);
+        RowManager rowMgr = databaseClient.newRowManager();
+        RawPlanDefinition maxConstraintValueQuery = rowMgr.newRawPlanDefinition(new StringHandle(currentMaxValueQuery));
+        JacksonHandle handle = new JacksonHandle().withFormat(Format.JSON).withMimetype("application/json");
+        handle.setPointInTimeQueryTimestamp(queryStartTimeInMillis);
+        JacksonHandle result = rowMgr.resultDoc(maxConstraintValueQuery, handle);
+        String newMaxConstraintValue = "";
+        try {
+            newMaxConstraintValue = result.get().get("rows").get(0).get("constraint").get("value").asText();
+            newMaxConstraintValue = newMaxConstraintValue.replaceAll("[\"']", "");
+        } catch (Exception ex) {
+            logger.warn("Failed to get a valid Maximum Constraint value: " + ex.getMessage());
+        }
+        return newMaxConstraintValue;
+    }
+
+    private String buildMaxValueSerializedQuery(String currentQuery, String constraintColumn) {
+        String orderByDescendingPhrase = "{\"ns\":\"op\", \"fn\":\"order-by\", \"args\":[{\"ns\":\"op\", \"fn\":\"desc\", \"args\":[\"" + constraintColumn + "\"]}]}";
+        try {
+            ObjectNode orderByDescendingNode = (ObjectNode) mapper.readTree(orderByDescendingPhrase);
+            String limitOnePhase = "{\"ns\":\"op\", \"fn\":\"limit\", \"args\":[1]}";
+            ObjectNode limitOneNode = (ObjectNode) mapper.readTree(limitOnePhase);
+            String constraintColumnPhase = "{\"ns\":\"op\", \"fn\":\"select\", \"args\":[[{\"ns\":\"op\", \"fn\":\"as\", \"args\":[\"constraint\", {\"ns\":\"op\", \"fn\":\"col\", \"args\":[\"" + constraintColumn + "\"]}]}]]}";
+            ObjectNode constraintColumnNode = (ObjectNode) mapper.readTree(constraintColumnPhase);
+            ObjectNode queryRoot = (ObjectNode) mapper.readTree(currentQuery);
+            ArrayNode rootArgsArray = (ArrayNode) queryRoot.get("$optic").get("args");
+            rootArgsArray.add(orderByDescendingNode);
+            rootArgsArray.add(limitOneNode);
+            rootArgsArray.add(constraintColumnNode);
+            return mapper.writeValueAsString(queryRoot);
+        } catch (JsonProcessingException e) {
```

Comment:
I think this method should declare `throws JsonProcessingException` and then the one calling method should include it in its try/catch so that any error involved in trying to get the max value is handled the same way (right now, this would be handled differently because it will propagate up to poll() and result in no rows being returned).